Sources/Sentry/Profiling/SentryContinuousProfiler.mm

```diff
@@ -159,14 +164,32 @@ + (void)scheduleTimer
 
 + (void)timerExpired
 {
-    std::lock_guard<std::mutex> l(_threadUnsafe_gContinuousProfilerLock);
-    if (![_threadUnsafe_gContinuousCurrentProfiler isRunning]) {
-        SENTRY_LOG_WARN(@"Current profiler is not running. Sending whatever data it has left "
-                        @"and disabling the timer from running again.");
-        disableTimer();
+    {
+        std::lock_guard<std::mutex> l(_threadUnsafe_gContinuousProfilerLock);
+        if (![_threadUnsafe_gContinuousCurrentProfiler isRunning]) {
+            SENTRY_LOG_WARN(@"Current profiler is not running. Sending whatever data it has left "
+                            @"and disabling the timer from running again.");
+            disableTimer();
+        }
+
+        _sentry_threadUnsafe_transmitChunkEnvelope();
+
+        if (!_stopCalled) {
+            return;
+        }
     }
 
-    _sentry_threadUnsafe_transmitChunkEnvelope();
+    [self stopTimerAndCleanup];
```

Comment:
`m`: Is it intended to first transmit the chunk envelope and then stop the profiler? Doesn't it make more sense to stop the profiler first and then transmit the chunk envelope?

Reply:
I think there's such a short amount of time between the two it doesn't much matter. And even less so if/when we shuffle the serialization/envelope capture off to a separate thread. The ordering is functionally the same as it was before this change: 
<img width="455" alt="image" src="https://github.com/user-attachments/assets/1640f4b7-b061-425d-a81d-5424a6e8ba46">